### PR TITLE
py-{boto3,botocore,jmespath,s3transfer} bump

### PR DESCRIPTION
**Pull request by haampie**
Bump spack's own deps

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-boto3/package.py [hunk 1]
```diff
@@ -12,24 +12,47 @@ class PyBoto3(PythonPackage):
...
     version("1.10.38", sha256="6cdb063b2ae5ac7b93ded6b6b17e3da1325b32232d5ff56e6800018d4786bba6")
+    version("1.9.253", sha256="d93f1774c4bc66e02acdda2067291acb9e228a035435753cb75f83ad2904cbe3")
     version("1.9.169", sha256="9d8bd0ca309b01265793b7e8d7b88c1df439737d77c8725988f0277bbf58d169")
 
+    depends_on("python@3.7:", when="@1.26:", type=("build", "run"))
     depends_on("python@3.6:", when="@1.18:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.6:", when="@1.17.27", type=("build", "run"))
-    depends_on("python@2.6:", when="@1.9:1.10", type=("build", "run"))
+    depends_on("python@2.7:2.8,3.6:", when="@1.17:", type=("build", "run"))
+    depends_on("python@2.6:", when="@1.9:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-boto3/package.py:36]: You could honestly remove all this, we only support Python 3.7+

File: var/spack/repos/builtin/packages/py-boto3/package.py [hunk 2]
```diff
@@ -12,24 +12,47 @@ class PyBoto3(PythonPackage):
...
+    depends_on("py-botocore@1.13.50:1.13", when="@1.10", type=("build", "run"))
+    depends_on("py-botocore@1.12.253:1.12", when="@1.9", type=("build", "run"))
+
+    depends_on("py-jmespath@0.7.1:0", when="@:1.20", type=("build", "run"))
+    depends_on("py-jmespath@0.7.1:1", type=("build", "run"))
+
+    depends_on("py-s3transfer@0.6.0:0.6", when="@1.24:", type=("build", "run"))
+    depends_on("py-s3transfer@0.5.0:0.5", when="@1.18:1.23", type=("build", "run"))
+    depends_on("py-s3transfer@0.3.0:0.3", when="@1.17", type=("build", "run"))
     depends_on("py-s3transfer@0.2.0:0.2", when="@:1.10", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-boto3/package.py]:
```suggestion
    depends_on("py-s3transfer@0.6", when="@1.24:", type=("build", "run"))
    depends_on("py-s3transfer@0.5", when="@1.18:1.23", type=("build", "run"))
    depends_on("py-s3transfer@0.3", when="@1.17", type=("build", "run"))
    depends_on("py-s3transfer@0.2", when="@:1.10", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-botocore/package.py [hunk 3]
```diff
@@ -12,20 +12,35 @@ class PyBotocore(PythonPackage):
...
     version("1.13.44", sha256="a4409008c32a3305b9c469c5cc92edb5b79d6fcbf6f56fe126886b545f0a4f3f")
     version("1.13.38", sha256="15766a367f39dba9de3c6296aaa7da31030f08a0117fd12685e7df682d8acee2")
+    version("1.12.253", sha256="3baf129118575602ada9926f5166d82d02273c250d0feb313fc270944b27c48b")
     version("1.12.169", sha256="25b44c3253b5ed1c9093efb57ffca440c5099a2d62fa793e8b6c52e72f54b01e")
 
     depends_on("python@2.6:", when="@1.12:1.13", type=("build", "run"))
     depends_on("python@2.7:", when="@1.19", type=("build", "run"))
     depends_on("python@2.7:2.8,3.6:", when="@1.20", type=("build", "run"))
     depends_on("python@3.6:", when="@1.21:", type=("build", "run"))
+    depends_on("python@3.7:", when="@1.26:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-botocore/package.py:39]: Same here, could remove

**Review by adamjstewart — DISMISSED** [review]
All suggestions are very much optional, feel free to merge if you don't care.

File: var/spack/repos/builtin/packages/py-boto3/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/py-boto3/package.py:36]: Hm, I thought to leave it since it's a spack dependency.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**